resolwe_bio/processes/workflows/rnaseq_variant_calling.py

```diff
@@ -0,0 +1,400 @@
+                "other selection criteria. Valid types are INDEL, SNP, MIXED, MNP, SYMBOLIC, "
+                "NO_VARIATION. Can be specified multiple times.",
+                default=["SNP"],
+            )
+
+        class Advanced:
+            """Advanced options."""
+
+            java_gc_threads = IntegerField(
+                label="Java ParallelGCThreads",
+                default=2,
+                description="Sets the number of threads used during parallel phases of "
+                "the garbage collectors.",
+            )
+            max_heap_size = IntegerField(
+                label="Java maximum heap size (Xmx)",
+                default=12,
+                description="Set the maximum Java heap size (in GB).",
+            )
+
+        bbduk = GroupField(
+            Bbduk, label="Preprocessing with BBDuk", hidden="!preprocessing"
+        )
+        alignment = GroupField(
+            Alignment,
+            label="Alignment with STAR",
+        )
+        haplotype_caller = GroupField(
+            HaplotypeCaller, label="Options for HaplotypeCaller"
+        )
+        select_variants = GroupField(
+            SelectVariants,
+            label="Options for GATK SelectVariants",
+            hidden="!exclude_filtered",
+        )
+        advanced = GroupField(Advanced, label="Advanced options")
+
+    class Output:
+        """Output fields."""
+
+        # Workflow do not have outputs.
+
+    def run(self, inputs, outputs):
+        """Run the workflow."""
+
+        if inputs.preprocessing:
+
+            input_preprocessing = {
+                "reads": inputs.reads,
+                "reference": {
+                    "sequences": inputs.bbduk.adapters or [],
+                    "literal_sequences": inputs.bbduk.custom_adapter_sequences,
+                },
+                "operations": {
+                    "quality_encoding_offset": inputs.bbduk.quality_encoding_offset,
+                    "ignore_bad_quality": inputs.bbduk.ignore_bad_quality,
+                },
+            }
+
+            if inputs.reads.type.startswith("data:reads:fastq:single:"):
+                slug_bbduk = "bbduk-single"
+            elif inputs.reads.type.startswith("data:reads:fastq:paired:"):
+                slug_bbduk = "bbduk-paired"
+            else:
+                self.error("Wrong reads input type.")
+
+            preprocessing = Data.create(
+                process=BioProcess.get_latest(slug=slug_bbduk),
+                input=input_preprocessing,
+            )
+
+        input_star = {
+            "genome": inputs.genome,
+            "two_pass_mapping": {
+                "two_pass_mode": inputs.alignment.two_pass_mode,
+            },
+            "output_options": {
+                "out_unmapped": inputs.alignment.out_unmapped,
+            },
+        }
+
+        if inputs.preprocessing:
+            input_star["reads"] = preprocessing
+        else:
+            input_star["reads"] = inputs.reads
```

Comment:
This could be simplified and added to the dictionary above as `"reads": preprocessing if inputs.preprocessing else inputs.reads`.